src/linter/ui5Types/SourceFileLinter.ts

```diff
@@ -449,6 +460,164 @@ export default class SourceFileLinter {
+				} else if (classMember.name?.getText() === "metadata") {
+					componentManifest = classMember;
+				}
+			});
+		}
+
+		let hasAsyncInterface = false;
+		if (classInterfaces && ts.isPropertyDeclaration(classInterfaces) &&
+			classInterfaces.initializer && ts.isArrayLiteralExpression(classInterfaces.initializer)) {
+			hasAsyncInterface = classInterfaces.initializer
+				.elements.some((implementedInterface) =>
+					implementedInterface.getText() === "\"sap.ui.core.IAsyncContentCreation\"");
+		}
+
+		let manifestJson: propsRecord = {};
+		if (componentManifest && ts.isPropertyDeclaration(componentManifest) &&
+			componentManifest.initializer && ts.isObjectLiteralExpression(componentManifest.initializer)) {
+			manifestJson = this.extractPropsRecursive(componentManifest.initializer) ?? {};
+		}
+
+		let rootViewAsyncFlag: boolean | undefined;
+		let routingAsyncFlag: boolean | undefined;
+		let rootViewAsyncFlagNode: ts.Node | undefined;
+		let routingAsyncFlagNode: ts.Node | undefined;
+
+		if (manifestJson.manifest?.value === "\"json\"") { // The manifest is an external manifest.json file
+			const parsedManifestContent =
+				JSON.parse(this.#manifestContent ?? "{}") as SAPJSONSchemaForWebApplicationManifestFile;
+
+			const {rootView, routing} = parsedManifestContent["sap.ui5"] ?? {} as JSONSchemaForSAPUI5Namespace;
+			// @ts-expect-error async is part of RootViewDefFlexEnabled and RootViewDef
+			rootViewAsyncFlag = rootView?.async as boolean;
+			routingAsyncFlag = routing?.config?.async;
+		} else {
+			/* eslint-disable @typescript-eslint/no-explicit-any */
+			const instanceOfPropsRecord = (obj: any): obj is propsRecord => {
+				return !!obj && typeof obj === "object";
+			};
+
+			let manifestSapui5Section: propsRecord | undefined;
+			if (instanceOfPropsRecord(manifestJson.manifest?.value) &&
+				instanceOfPropsRecord(manifestJson.manifest.value["\"sap.ui5\""].value)) {
+				manifestSapui5Section = manifestJson.manifest.value["\"sap.ui5\""].value;
+			}
+
+			if (instanceOfPropsRecord(manifestSapui5Section) &&
+				instanceOfPropsRecord(manifestSapui5Section?.rootView?.value) &&
+				typeof manifestSapui5Section?.rootView?.value.async?.value === "boolean") {
+				rootViewAsyncFlag = manifestSapui5Section?.rootView?.value.async?.value;
+				rootViewAsyncFlagNode = manifestSapui5Section?.rootView?.value.async?.node;
+			}
+
+			if (instanceOfPropsRecord(manifestSapui5Section) &&
+				instanceOfPropsRecord(manifestSapui5Section?.routing?.value) &&
+				instanceOfPropsRecord(manifestSapui5Section?.routing?.value.config?.value) &&
+				typeof manifestSapui5Section?.routing?.value.config?.value.async?.value === "boolean") {
+				routingAsyncFlag = manifestSapui5Section?.routing?.value.config?.value.async?.value;
+				routingAsyncFlagNode = manifestSapui5Section?.routing?.value.config?.value.async?.node;
+			}
+		}
+
+		if (!hasAsyncInterface) {
+			if (rootViewAsyncFlag !== true || routingAsyncFlag !== true) {
```

Comment:
If there is no rootView or routing defined, the values must not be checked. Both sections are optional. This currently causes false-positives.

Reply:
Thanks!
Fixed it